src/core/server/saved_objects/types.ts

```diff
@@ -111,7 +112,15 @@ export interface SavedObjectsFindOptions {
   /** An optional OpenSearch preference value to be used for the query **/
   preference?: string;
   workspaces?: string[];
-  queryDSL?: Record<string, any>;
+  /**
+   * The params here will be combined with bool clause and is used for filtering with ACL structure.
+   */
+  ACLSearchParams?: {
+    workspaces?: string[];
+    principals?: Principals;
+    workspacePermissionModes?: string[];
+    objectPermissionModes?: string[];
```

Comment:
Could we merge `workspacePermissionModes` and `objectPermissionModes` into one permission modes object?

Reply:
As now the permissionModes here is a "or" combination, I think we can merge these two modes into one.

Reply:
Is that means we can't merge these two modes into one?

Reply:
We can, will do that

Reply:
Done.